enhancements/olm-admission-webhooks.md

```diff
@@ -0,0 +1,134 @@
+type Webhook struct {
+	Name                    string                                          `json:"name"`
+	Rules                   []admissionregistrationv1.RuleWithOperations    `json:"rules"`
+	FailurePolicy           *admissionregistrationv1.FailurePolicyType      `json:"failurePolicy,omitempty"`
+	MatchPolicy             *admissionregistrationv1.MatchPolicyType        `json:"matchPolicy,omitempty"`
+	NamespaceSelector       *metav1.LabelSelector                           `json:"namespaceSelector,omitempty"`
+	ObjectSelector          *metav1.LabelSelector                           `json:"objectSelector,omitempty"`
+	SideEffects             *admissionregistrationv1.SideEffectClass        `json:"sideEffects"`
+	TimeoutSeconds          *int32                                          `json:"timeoutSeconds,omitempty"`
+	AdmissionReviewVersions []string                                        `json:"admissionReviewVersions"`
+	ReinvocationPolicy      *admissionregistrationv1.ReinvocationPolicyType `json:"reinvocationPolicy,omitempty"`
+}
+
+// WebhookDescription is used to provide OLM with information about the Webhook being shipped with the operator
+type WebhookDescription struct {
+	// Type is used to identify the admission webhook as a validating or mutating webhook. Accepted values include `validating` and `mutating`
+	Type                    WebhookAdmissionType                            `json:"type"`
+	// DeploymentName must match one of the deployments included in the CSV.
+	DeploymentName          string                                          `json:"deploymentName,omitempty"`
+	// ContainerPort specifies which port the service should forward requests to.
+	ContainerPort           int32                                           `json:"containerPort,omitempty"`
+	// Webhook contains the information that must be sent to the validating or mutating webhook. 
+	Webhook                 Webhook                                      `json:"webhook"`
+}
+```
+
+### Securing webhook via certificates
+
+The certificate creation and lifecycle management is planned to utilize the same certs package that is already in OLM managing the certificates for api services. OLM has tested this feature extensively at this point and could easily port it to Webhooks.
+
+### Integration with Cert Management tools
+
+There is a great deal of community support for integration with certificate management projects such as the [service-ca operator](https://github.com/openshift/service-ca-operator) or [cert-manager](https://github.com/jetstack/cert-manager). Cluster Admins should be given the opportunity to integrate with either of these tools be disabling the OLM management of certs and adding appropriate annotations to the deployment and webhook resources via the [SubscriptionConfig resource](https://github.com/operator-framework/operator-lifecycle-manager/blob/8985872bfd5888c8d7d49e4dfc9be162a87691e7/pkg/api/apis/operators/v1alpha1/subscription_types.go#L41-L87).
```

Comment:
> disabling the OLM management of certs

Globally, or per-resource?

Reply:
Ah - subscriptions are typically per resource and was what I had planned.